### PR TITLE
Make JBoss Java Release profiles not active by default

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] native</b>
</details>

### CODE REVIEW

**Comment by triceo** [comment]
Does this fix the issue for you?
Both Radovan and I tried that earlier today, and it had no effect on us.

**Comment by Christopher-Chianelli** [comment]
@triceo I disable a lot more profiles. When I switch to master, I get errors. When I switch back to this branch, errors goes away after the import maven project process complete

**Comment by Christopher-Chianelli** [comment]
(my testing process was: quit IDEA; switch to branch; open IDEA; wait for import process to complete)

**Comment by rsynek** [comment]
For me, the single profile helped. Now the question is if the deactivation of all the suspicious profiles helps everybody else. @triceo would you give it a try, please?

**Review by triceo — CHANGES_REQUESTED** [review]
Sorry, but this simply doesn't work for me.

Do note that Quarkus advice to make IDEA work is to disable those profiles in the IDE directly - and it doesn't work for me even when I do. I disabled all the profiles, and still got the issue.

There has to be some other variable which we are not aware of, which causes this to work for some people, but not for others.

**Review by triceo — COMMENTED** [review]
So, I finally got it working. After all of these steps:

- Purge every IML file from the project.
- Manually go over every file in `.idea` folder and remove all references to JDK 8.
- Clean up local Maven repo just to be sure.

All in all, my answer here is that this PR will help, as long as you are checking out a clean project.
If you've already got IntelliJ in some random working state, this may or may not help.

**Comment by rsynek** [comment]
My question is: is this a minimal set of profiles we need to deactivate?

**Comment by triceo** [comment]
> My question is: is this a minimal set of profiles we need to deactivate?

Honestly, I do not want to go through the process of finding out. This was a nightmare enough already.

**Comment by rsynek** [comment]
> > My question is: is this a minimal set of profiles we need to deactivate?
> 
> Honestly, I do not want to go through the process of finding out. This was a nightmare enough already.
> I do not have confidence that I know all the necessary steps to make this determination reliably and reproducibly.

We probably don't have to; Quarkus suggests it's just two of them:
- include-jdk-misc
- compile-java8-release-flag

You've already checked that turning these off in IntelliJ fixes the issue.

**Comment by triceo** [comment]
> You've already checked that turning these off in IntelliJ fixes the issue.

Have I? The only thing I can reliably say is that this PR does the job. When I was switching profiles off yesterday, it had no effect - probably because I did not do the steps I described above. And due to the insanity of those steps, I am not really excited about doing these experiments again.

**Comment by rsynek** [comment]
> > You've already checked that turning these off in IntelliJ fixes the issue.
> 
> Have I? The only thing I can reliably say is that this PR does the job. When I was switching profiles off yesterday, it had no effect - probably because I did not do the steps I described above. And due to the insanity of those steps, I am not really excited about doing these experiments again.

Sorry, I understood that the exact workaround Quarkus recommends worked for you. For me it did.

File: pom.xml [hunk 1]
```diff
@@ -139,6 +139,78 @@
         <module>optaplanner-distribution</module>
       </modules>
     </profile>
+
+    <!-- Disable Java 8 active by default profiles from JBoss -->
```
Comment (ge0ffrey) [inline pom.xml:143]: Let's add a expiration condition comment, at which point all these extra profiles can be removed.

Comment (ge0ffrey) [reply pom.xml:143]: Please create an issue upstream on jboss-parent if that hasn't been done yet, and link it here.

**Comment by triceo** [comment]
@Christopher-Chianelli Can you please resolve the comments on this PR, so that we can merge it? 
I only just now noticed it isn't yet.

**Review by triceo — APPROVED** [review]

**Pull request merged**